src/modROS.lua

```diff
@@ -136,112 +130,55 @@ function ModROS:update(dt)
 end
 
 
--- -- A.1 getPublisher - instantiate publishers for each spec instance, called from RosVehicle.lua
--- function ModROS:getPublisher(topic_name, topic_type)
---     local pub = Publisher.new(self._conx, topic_name, topic_type)
---     return pub
--- end
-
-
--- B.1 sim_time publisher: publish the farmsim time
+-- A.1 sim_time publisher: publish the farmsim time
 function ModROS:publish_sim_time_func()
     local msg = rosgraph_msgs_Clock.new()
     msg.clock = ros.Time.now()
     self._pub_clock:publish(msg)
 end
 
 
--- B.2. odom publisher
+-- A.2. odom publisher
 -- a function to publish ground-truth Pose and Twist of all vehicles (including non-drivable) based on their in-game position and orientation
-function ModROS:publish_veh_func()
+function ModROS:publish_veh_odom_func()
     -- FS time is "frozen" within a single call to update(..), so this
     -- will assign the same stamp to all Odometry messages
     local ros_time = ros.Time.now()
     for _, vehicle in pairs(g_currentMission.vehicles) do
+        -- only if the vehicle is spec_rosVehicle, the pubOdom() can be called
```

Comment:
Can be called, or *is* called?

Should we be filtering for `rosVehicle` specs?

Reply:
Same question below everywhere the same comment occurs.